app/uk/gov/hmrc/apiplatformmicroservice/apidefinition/services/ApiDefinitionsForCollaboratorFetcher.scala

```diff
@@ -17,23 +17,34 @@
 package uk.gov.hmrc.apiplatformmicroservice.apidefinition.services
 
 import javax.inject.{Inject, Singleton}
-import uk.gov.hmrc.apiplatformmicroservice.apidefinition.connectors.ApiDefinitionConnector
 import uk.gov.hmrc.apiplatformmicroservice.apidefinition.models.{APIDefinition, APIStatus, APIVersion, PrivateApiAccess}
+import uk.gov.hmrc.apiplatformmicroservice.common.Recoveries
 import uk.gov.hmrc.apiplatformmicroservice.thirdpartyapplication.services.ApplicationIdsForCollaboratorFetcher
 import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.Future.successful
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class ApiDefinitionsForCollaboratorFetcher @Inject()(apiDefinitionConnector: ApiDefinitionConnector, appIdsFetcher: ApplicationIdsForCollaboratorFetcher)
-                                                    (implicit ec: ExecutionContext) {
+class ApiDefinitionsForCollaboratorFetcher @Inject()(principalDefinitionService: PrincipalApiDefinitionService,
+                                                     subordinateDefinitionService: SubordinateApiDefinitionService,
+                                                     appIdsFetcher: ApplicationIdsForCollaboratorFetcher)
+                                                    (implicit ec: ExecutionContext) extends Recoveries {
 
   def apply(email: Option[String])(implicit hc: HeaderCarrier): Future[Seq[APIDefinition]] = {
+    val principalDefinitionsFuture = principalDefinitionService.fetchAllDefinitions
+    val subordinateDefinitionsFuture  = subordinateDefinitionService.fetchAllDefinitions recover recoverWithDefault(Seq.empty[APIDefinition])
+
     for {
-      allApiDefinitions <- apiDefinitionConnector.fetchAllApiDefinitions
+      principalDefinitions <- principalDefinitionsFuture
+      subordinateDefinitions <- subordinateDefinitionsFuture
+      combinedDefinitions = combineDefinitions(principalDefinitions, subordinateDefinitions)
       applicationIds <- email.fold(successful(Seq.empty[String]))(appIdsFetcher(_))
```

Comment:
Might not need (_) on this

Reply:
Yeah it looks weird but it doesn't work without it. I think it's because it's not a function, but an instance of a class. And here we're calling the apply method on that instance.